### PR TITLE
Bundle busybox, fix #211

### LINKED ISSUE #211: Use bundled busybox as Toybox replacement
toybox doesn't offer awk, but opengapps installer uses it.

### PR DESCRIPTION
**Pull request by klausenbusk**
We only use the busybox "binaries", not any of the system binaries.
This also remove xz-binary, as busybox include xz.

Not tested..

### CODE REVIEW

**Comment by mfonville** [comment]
Also we should remove lines:
https://github.com/opengapps/opengapps/blob/master/scripts/inc.installdata.sh#L105
https://github.com/opengapps/opengapps/blob/master/scripts/inc.installdata.sh#L106
https://github.com/opengapps/opengapps/blob/master/scripts/inc.installdata.sh#L107
https://github.com/opengapps/opengapps/blob/master/scripts/inc.installdata.sh#L534
https://github.com/opengapps/opengapps/blob/master/scripts/inc.installdata.sh#L535
https://github.com/opengapps/opengapps/blob/master/scripts/inc.installdata.sh#L536
(and update wiki accordingly)

**Comment by mfonville** [comment]
Note to myself @mfonville: If merging this I need to update the license files used in the buildbot.

**Comment by klausenbusk** [comment]
Most of what you said should be fixed now.

File: scripts/inc.updatebinary.sh [hunk 1]
```diff
@@ -18,12 +18,18 @@ echo '#!/sbin/sh
...
+if [ -z "$READY" ]; then
+  unzip -o "$3" '"$EXTRACTFILES"' -d /tmp
+
+  chmod +x /tmp/busybox
+  install -d /tmp/bin
+  bb=/tmp/busybox
+  for i in $($bb --list); do
+    ln -s $bb /tmp/bin/$i
+  done
+  READY=true PATH="/tmp/bin" $bb ash "$@"
```
Comment (klausenbusk) [inline scripts/inc.updatebinary.sh]: What should we call this env variable?

File: scripts/inc.packagetarget.sh [hunk 2]
```diff
@@ -55,13 +55,13 @@ aromaupdatebinary() {
...
-    arm*) xzbin="xzdec-arm";;
-    x86*) xzbin="xzdec-x86";;
+bundlebusybox() {
+  case "$ARCH" in #Include official busybox binary
+    arm*) busyboxbin="busybox-armv7l";;
+    x86*) busyboxbin="busybox-i686";;
   esac
-  copy "$SCRIPTS/xz-resources/$xzbin" "$build/xzdec"
-  EXTRACTFILES="$EXTRACTFILES xzdec"
+  copy "$SCRIPTS/busybox-resources/$busyboxbin" "$build/busybox"
```
Comment (klausenbusk) [inline scripts/inc.packagetarget.sh]: I removed the `chmod +x "$build/busybox"`, as I have marked the busybox executable in git.

**Comment by mfonville** [comment]
@klausenbusk I will check it :)
I can already comment on the busybox LICENSE file; it is not necessary to add the full file, you can just mention the applicable license (GPLv2) in the howto and a link to their website (where they can find the exact license and terms) :)

**Comment by mfonville** [comment]
Another note: the filename HOWTO has a typo

**Comment by mfonville** [comment]
ok; I am done with my comments. We are almost there I guess :-)

File: scripts/inc.updatebinary.sh (hunk 1 above)
Comment (mfonville) [inline scripts/inc.updatebinary.sh]: see my hint how we could do it without an extra variable :)

File: scripts/inc.packagetarget.sh (hunk 2 above)
Comment (mfonville) [inline scripts/inc.packagetarget.sh]: aye

**Comment by klausenbusk** [comment]
Ready for round 3

**Comment by mfonville** [comment]
Super :-) LGTM
I will merge it into a new branch for the testing phase; I will try it on my non-CM13 recovery device to see if it works.
If you could verify whether it works on your CM13 recovery :-)

Should also give you the chance to still get that /cache/log/recovery file ;-)
I would like to merge the CM13-recovery recognizing together when merging this because it would allow us to troubleshoot properly.

File: scripts/inc.updatebinary.sh [hunk 3]
```diff
@@ -18,12 +18,21 @@ echo '#!/sbin/sh
...
-  unzip -o "$3" "$f" -d /tmp;
-done'> "$build/META-INF/com/google/android/update-binary"
-case "$EXTRACTFILES" in
-  *xzdec*) echo 'chmod +x /tmp/xzdec'>> "$build/META-INF/com/google/android/update-binary";; #xz-decompression binary bundled
-esac
+case "$PATH" in
+  */tmp/bin/*)
+    ;;
+  *)
+    unzip -o "$3" '"$EXTRACTFILES"' -d /tmp
```
Comment (mfonville) [inline scripts/inc.updatebinary.sh]: I was very dumb; this is of course executed in the non-busybox environment, so should still be done old-style with the for-loop. I fixed this in the branch that I will push soon.

Comment (klausenbusk) [inline scripts/inc.updatebinary.sh]: Done.

**Comment by mfonville** [comment]
I pushed my updated branch

**Comment by mfonville** [comment]
I can confirm it works on good ol' TWRP

**Pull request closed without merging**